### DISCUSSION TITLE
Enabling manual mesh with Ender 3 v2

### DISCUSSION THREAD

**Discussion by bsidebottom** [Jyers/Marlin#1197] [Q&A] [answered]
Hey all,

I have searched high and low and cannot find the answers. I need to compile my own version to enable the BTT smart runout sensor. This works fine and have that all working.

Now I want to add in the manual mesh that the precompiled versions have. I have tried uncommenting the parts I need (manual mesh, lcd screen etc.) but I then get errors and all the posts around the web say the Ender 3 v2 screen cannot be modified and doesn't work for the menu items.

So how come the precompiled version works fine and I have the menu on my screen, but compiling my own, I cannot get it to work? I have tried enabling and disabling different parts but cannot get a combination that compiles and shows the option on my screen WITH the BTT smart sensor parts enabled (which I really need).

Sorry if this has been asked before but I couldn't find it here or on any other sources (YouTube, other forums etc.).

Many thanks for any help! Love this Jyers version though. Custom colours and all the options at my fingertips in screen is an amazing upgrade!! :)

**Comment by Cal-LF** [comment, marked as answer]
Not sure what template you are using. Make sure you are starting with Ender3V2JyresUI default branch. Copy from the E3V2 Templates the Manual Mesh 3X3 config files to the Marlin folder, replacing the existing 2 config files.
In VS Code enable the motion sensor.  1482 #define FILAMENT_MOTION_SENSOR
You can change other things like the grid size for manual mesh.  1593 #define GRID_MAX_POINTS_X 3, 4 or 5

[Compiling Steps.txt](https://github.com/Jyers/Marlin/files/6864472/Compiling.Steps.txt)

**Reply by bsidebottom** [reply]
Aha!! Amazing, didn't know about the configs. Thank you so much, will give this a go tomorrow :) Really appreciate it!

**Reply by bsidebottom** [reply]
Spot on, working great, thank you!
